nmigen_boards/chameleon96.py

```diff
@@ -0,0 +1,100 @@
+import os
+import subprocess
+
+from nmigen import *
+from nmigen.build import *
+from nmigen.vendor.intel import *
+from .resources import *
+
+
+__all__ = ["Chameleon96Platform"]
+
+class Chameleon96Platform(IntelPlatform):
+    device      = "5CSEBA6" # Cyclone V SE 110K LEs
+    package     = "U19"     # UBGA-484
+    speed       = "I7"
+    default_clk = "cyclonev_oscillator"
+    resources   = [
+        # WIFI and BT LEDs
+        *LEDResources(
+            pins="Y19 Y20",
+            attrs=Attrs(io_standard="2.5 V")),
+
+        # TDA19988 HDMI transmitter
+        Resource("tda19988", 0,
```

Comment:
So, you're making a fairly reasonable choice of grouping related pins for multiple different protocols together under one resource. Unfortunately I think they need to be split. The reason is that our current system is heavily tailored towards that in two ways:
  * We standardize on resource layout by using resource factory functions. Unfortunately we don't have subsignal factory functions, which means that any resources-as-subsignals will not benefit from additional checks and consistency guarantees these provide.
  * We allow overriding parameters (direction and XDR) when you request a resource. Unfortunately this means that you can't e.g. let some function or block or class use custom overrides with an I2C resource-as-subsignal.

So I think you should split these into `tda19988_i2c`, `tda19988_i2s` and so on. Arguably this is a design issue in nMigen but for now it's the least bad option.

Reply:
Thanks for the explanation. I pushed an update and hope I understood you correctly.